### PR TITLE
Support Django 5.0 and Python 3.12, drop Python 3.8

### PR DESCRIPTION
**Pull request by adamchainz**
Copied most of the approach from 35d10aadf1faabbbe465747783e14e2075c3a137. I don’t see any new warnings on Python 3.12 + Django 5.0 when running locally.

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -15,8 +15,19 @@ jobs:
...
+        - '4.1'
+        - '4.2'
+        - '5.0'
+        python_version:
+        - '3.7'
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
```
Comment (adamchainz) [inline .github/workflows/python-app.yml:30]: splitting one per line will reduce the diff noise when adding/removing versions in the future

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -15,8 +15,19 @@ jobs:
...
-        django_version: [ '3.2' , '4.0', '4.1', '4.2']
-        python_version: [ '3.7', '3.8', '3.9' , '3.10', '3.11' ]
+        django_version:
+        - '3.2'
+        - '4.0'
+        - '4.1'
+        - '4.2'
+        - '5.0'
+        python_version:
+        - '3.7'
```
Comment (adamchainz) [inline .github/workflows/python-app.yml:25]: I'd also be happy to drop Python 3.7 support in this PR, since it’s EOL

File: quicktest.py [hunk 3]
```diff
@@ -71,6 +71,7 @@ def run_tests(self):
                     'context_processors': [
                         'django.contrib.auth.context_processors.auth',
                         'django.contrib.messages.context_processors.messages',
+                        'django.template.context_processors.request',
```
Comment (adamchainz) [inline quicktest.py:74]: Fixes this system check warning:

```
?: (admin.W411) 'django.template.context_processors.request' must be enabled in DjangoTemplates (TEMPLATES) in order to use the admin navigation sidebar.
```

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (Gagaro) [reply .github/workflows/python-app.yml:25]: Sure, feel free to remove it.

**Comment by Gagaro** [comment]
Thanks for the PR, I'll merge after you remove the support of Python 3.7 then.

**Comment by adamchainz** [comment]
Done.

**Pull request merged**